pkg/sfu/forwarder.go

```diff
@@ -1734,3 +1757,40 @@ done:
 
 	return float64(distance) / float64(maxSeenLayer.Temporal+1)
 }
+
+func getNextTimestamp(lastTS uint32, refTS uint32, expectedTS uint32) (uint32, string) {
+	isInOrder := func(val1, val2 uint32) bool {
+		diff := val1 - val2
+		return diff != 0 && diff < (1<<31)
+	}
+
+	rl := isInOrder(refTS, lastTS)
+	el := isInOrder(expectedTS, lastTS)
+	er := isInOrder(expectedTS, refTS)
+
+	nextTS := lastTS + 1
+	explain := "l = r = e"
+
+	switch {
+	case rl && el && er: // lastTS < refTS < expectedTS
+		nextTS = uint32(float64(refTS) + 0.95*float64(expectedTS-refTS))
+		explain = fmt.Sprintf("l < r < e, %d, %d", refTS-lastTS, expectedTS-refTS)
```

Comment:
This is an interesting case. Using 95% towards expected because of time warp seen. When muting/unmuting audio, instead of applying mute/unmute on the track, if the track is paused and replaced with an empty track, there is time warp. There are time windows where wall clock moves by 5 seconds, but RTP time stamp moves only by 3 seconds worth or so. So, on an unmute, using the publisher's time stamp as is mean there is significant drift. Not exactly sure why that happens when replacing with an empty track, but that drift could be significant.